invenio_indexer/utils.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Utility functions for data processing."""
+
+from flask import current_app
+from invenio_records.models import RecordMetadata
+
+from .api import RecordIndexer
+
+
+def process_models_committed_signal(sender, changes):
+    """Handler for indexing record metadata."""
+    record_indexer = RecordIndexer()
+    op_map = {
+        'insert': 'index',
+        'update': 'index',
+        'delete': 'delete',
+    }
+    with record_indexer.create_producer() as producer:
+        for obj, change in changes:
+            if isinstance(obj, RecordMetadata):
+                if change in op_map:
+                    index, doc_type = record_indexer._record_to_index(
```

Comment:
As you suggested it's probably a good idea to make the ``_record_to_index`` into a public method.